mono/metadata/opcodes.h

```diff
@@ -20,21 +20,23 @@ MONO_BEGIN_DECLS
 
 #define MONO_CUSTOM_PREFIX 0xf0
 
+// FIXME duplication
+#ifndef MonoOpcodeEnum
+#define MonoOpcodeEnum MonoOpcodeEnum
+
 #define OPDEF(a,b,c,d,e,f,g,h,i,j) \
 	MONO_ ## a,
 
-#ifndef __MONO_METADATA_BASIC_BLOCK_H__
```

Comment:
I don't understand what this was for - the comment in `mono-basic-block.h` says 
```c
/*This function is here because opcodes.h is a public header*/

#ifndef __MONO_METADATA_OPCODES_H__
typedef enum _MonoOpcodeEnum MonoOpcodeEnum;
#endif

int
mono_opcode_value_and_size (const unsigned char **ip, const unsigned char *end, MonoOpcodeEnum *value);
```

Which sounds like they just want to `#include <opcodes.h>` in `mono-basic-block.h` but couldn't for some reason.

Reply:
Oh, but also: I don't understand why it's necessary to change this file for this PR.

Reply:
1. Cannot forward reference an enum, alas.
2. I think maybe no common header, or varying include order. This is the smaller but maybe not prettiest way. I can try adding  an include.
Could use ints more also, granted.

Reply:
ah, no forward ref for enum makes sense.